### PR TITLE
Update docker to use new entrypoint style

### PR DESCRIPTION
**Pull request by jaspervdj-luminal**
Since the input file is no longer specified first, this allows us to add
arguments without having to overwrite the entrypoint.  This does mean that
"simple" usage now requires the `-` parameter:

    docker run -i regula:latest - <some_file.cfn

I think this is outweighed by the advantages: this makes it easier to pass in
other arguments, without having to re-specify `/opt/regula/`.  For example:

    docker run --rm \
        --volume [HCL_DIRECTORY]:/workspace \
        fugue/regula \
        /workspace

Where we needed the following monstrosity before:

    docker run --rm \
        --entrypoint regula \
        --volume [HCL_DIRECTORY]:/workspace \
        fugue/regula \
        /workspace \
        /opt/regula

### CODE REVIEW

**Review by jason-fugue — APPROVED** [review]
Thanks! I'm in favor of including this in 0.7.0.

File: README.md [hunk 1]
```diff
@@ -104,18 +104,19 @@ Regula is available as a Docker image on DockerHub [here](https://hub.docker.com
...
-    -e AWS_ACCESS_KEY_ID=XXXXXX \
-    -e AWS_SECRET_ACCESS_KEY=XXXXXX \
-    -e AWS_DEFAULT_REGION=xx-xxxx-x \
-    fugue/regula /workspace /opt/regula
+    docker run --rm \
+        --volume [HCL_DIRECTORY]:/workspace \
+        -e AWS_ACCESS_KEY_ID=XXXXXX \
+        -e AWS_SECRET_ACCESS_KEY=XXXXXX \
+        -e AWS_DEFAULT_REGION=xx-xxxx-x \
+        fugue/regula /workspace
```
Comment (becki-at-luminal) [inline README.md:119]: Should this be `fugue/regula:v0.7.0 /workspace`?

Comment (jaspervdj-luminal) [reply README.md:119]: I think it depends on if users want to put this in a script, or just try out the latest version right now.  I added a note about it.

Comment (becki-at-luminal) [reply README.md:119]: Makes sense, thanks!

**Pull request merged**